domain/secret/state/state.go

```diff
@@ -2590,25 +2797,26 @@ func (st State) getRevisionForObsolete(
 	outputType, result any,
 	appOwners domainsecret.ApplicationOwners,
 	unitOwners domainsecret.UnitOwners,
-	revisionUUIDs ...string,
+	revUUIDs ...string,
 ) error {
-	if len(revisionUUIDs) == 0 && len(appOwners) == 0 && len(unitOwners) == 0 {
+	if len(revUUIDs) == 0 && len(appOwners) == 0 && len(unitOwners) == 0 {
 		return nil
 	}
 
 	q := fmt.Sprintf(`
 SELECT
     %s
-FROM secret_revision sr`, selectStmt)
+FROM secret_revision_obsolete sro
+INNER JOIN secret_revision sr ON sr.uuid = sro.revision_uuid`, selectStmt)
 
 	var queryParams []any
 	var joins []string
 	conditions := []string{
-		"sr.obsolete = true",
+		"sro.obsolete = true",
 	}
-	if len(revisionUUIDs) > 0 {
-		queryParams = append(queryParams, obsoleteRevisionUUIDs(revisionUUIDs))
-		conditions = append(conditions, "AND sr.uuid IN ($obsoleteRevisionUUIDs[:])")
+	if len(revUUIDs) > 0 {
+		queryParams = append(queryParams, revisionUUIDs(revUUIDs))
+		conditions = append(conditions, "AND sr.uuid IN ($revisionUUIDs[:])")
 	}
 	if len(appOwners) > 0 && len(unitOwners) > 0 {
 		queryParams = append(queryParams, appOwners, unitOwners)
```

Comment:
I still think the stuff below would be a lot clearer using WITH to create app owner and unit owner temp tables and then inner joining to those tables from the main query, as is done in a couple of other places. At the least, we should use the same approach to select based on app and unit owners in this package.

Reply:
CTEs are materialized in SQLite. JOIN has better performance in this case.